packages/terra-worklist-data-grid/src/subcomponents/Row.jsx

```diff
@@ -0,0 +1,138 @@
+import styles from './Row.module.scss';
+import RowSelectionCell from './RowSelectionCell';
+import Cell from './Cell';
+import rowShape from '../proptypes/rowShape';
+import WorklistDataGridPropTypes from '../proptypes/WorklistDataGridPropTypes';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The row's position in the Grid. This is zero based.
+   */
+  rowIndex: PropTypes.number,
+  /**
+   * String that specifies height of the row. Any valid CSS width value is accepted.
+  */
+  height: PropTypes.string,
+  /**
+   * The information about the Row.
+   */
+  row: rowShape.isRequired,
+  /**
+   * Boolean indicating whether or not the DataGrid should allow entire rows to be selectable. An additional column will be
+   * rendered to allow for row selection to occur.
+   */
+  hasSelectableRows: PropTypes.bool,
+  /**
+   * Callback function that will be called when a row is selected. Parameters: `onRowSelect(rowId)`
+   */
+  onRowSelect: PropTypes.func,
+  /**
+   * All columns currently displayed.
+   */
+  displayedColumns: PropTypes.arrayOf(WorklistDataGridPropTypes.columnShape),
+  /**
+   * Callback function that will be called when a cell in the row is selected.
+   */
+  onCellSelect: PropTypes.func,
+  /**
+   * A number indicating which column represents the row header. This is zero based.
+   */
+  rowHeaderIndex: PropTypes.number,
+  /**
+   * Id of the column in the row that is selected.
+   */
+  selectedCellColumnId: PropTypes.string,
+  /**
+   * Column index of the column in that has focus.
+   */
+  tabStopColumnIndex: PropTypes.number,
+
+};
+
+const defaultProps = {
+  hasSelectableRows: false,
+};
+
+function Row(props) {
+  const {
+    rowIndex,
+    height,
+    hasSelectableRows,
+    row,
+    displayedColumns,
+    rowHeaderIndex,
+    onRowSelect,
+    onCellSelect,
+    selectedCellColumnId,
+    tabStopColumnIndex,
+  } = props;
+
+  const theme = useContext(ThemeContext);
+
+  const handleCellSelect = (rowIdColId, coordinates) => {
+    if (hasSelectableRows) {
+      if (onRowSelect) {
+        onRowSelect(rowIdColId.rowId, rowIndex, coordinates);
+      }
+    } else if (onCellSelect) {
+      onCellSelect(rowIdColId, coordinates);
+    }
+  };
+
+  const getCellData = (cellRowIndex, cellColumnIndex, cell, rowId) => {
+    const columnId = displayedColumns[cellColumnIndex].id;
+    const isRowHeader = cellColumnIndex === (rowHeaderIndex + hasSelectableRows ? 1 : 0);
+
+    return (
+      <Cell
+        rowId={rowId}
+        columnId={columnId}
+        rowIndex={cellRowIndex}
+        columnIndex={cellColumnIndex}
+        key={`${rowId}_${columnId}`}
+        isTabStop={tabStopColumnIndex === cellColumnIndex}
+        isSelected={!hasSelectableRows && selectedCellColumnId === columnId}
+        cell={cell}
+        isRowHeader={isRowHeader}
+        onCellSelect={handleCellSelect}
+      />
+    );
+  };
+
+  const columnIndexOffSet = hasSelectableRows ? 1 : 0;
```

Comment:
you can move this towards the top of the function to line 78 and reuse it on line 91:

https://github.com/cerner/terra-framework/blob/29c6db233cd8de3700026e556358a2f6d8770a97/packages/terra-worklist-data-grid/src/subcomponents/Row.jsx#L77-L78

https://github.com/cerner/terra-framework/blob/29c6db233cd8de3700026e556358a2f6d8770a97/packages/terra-worklist-data-grid/src/subcomponents/Row.jsx#L91

Reply:
Done